src/sqlfluff/core/rules/std/__init__.py

```diff
@@ -0,0 +1,20 @@
+# No docstring here as it would appear in the rules docs.
+# Rule definitions for the standard ruleset, dynamically imported from the directory.
+# noqa
+
+import os
+from importlib import import_module
+from glob import glob
+
+rules_path = os.path.join(os.path.abspath(os.path.dirname(__file__)), "L*.py")
+
+for module in sorted(glob(rules_path)):
+    # Manipulate the module path to extract the filename without the .py
+    rule_id = module.split("/")[-1][:-3]
+    rule_class_name = f"Rule_{rule_id}"
+    rule_class = getattr(
+        import_module(f"sqlfluff.core.rules.std.{rule_id}"), rule_class_name
+    )
+    # Add the rule_classes to the module namespace to be imported in
+    # sqlfluff/core/rules/__init__.py
+    globals()[rule_class_name] = rule_class
```

Comment:
This feels a _bit_ nicer? I know exec is considered bad, but it was more readable imo 😂

Reply:
Did you consider using metaclasses or one of the related techniques discussed [here](https://stackoverflow.com/questions/3915315/generic-metaclass-to-keep-track-of-subclasses) to discover all the rule classes? I think that may be simpler and cleaner. I believe it would also simplify the work for #164 by not tying rule discovery to a specific module/directory structure and naming convention.

Reply:
Not yet! I'll take a read and have a play.